File: testdata/bench/javascript/BenchmarkTest00013.js
// Vulnerable: SQL injection via execute with string interpolation
const express = require('express');
const db = require('./db');
const router = express.Router();
router.post('/test013', (req, res) => {
    const category = req.body.category;
    const sort = req.body.sort;
    db.execute(`SELECT * FROM products WHERE category = '${category}' ORDER BY ${sort}`)
        .then(results => res.json(results));
});
module.exports = router;

File: testdata/bench/rust/BenchmarkTest00013.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Vulnerable: SQL injection via variable indirection
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("id").unwrap_or(&String::new()).clone();
    let x = param.clone();
    let y = x;
    let sql = format!("SELECT * FROM accounts WHERE id = {}", y);
    // sqlx::query(&sql).execute(&pool).await
    HttpResponse::Ok().body(sql)
}
